server/client.go

```diff
@@ -2026,9 +2026,25 @@ func (c *client) processConnect(arg []byte) error {
 	}
 
 	// If websocket client and JWT not in the CONNECT, use the cookie JWT (possibly empty).
-	if ws := c.ws; ws != nil && c.opts.JWT == "" {
+	if ws := c.ws; ws != nil && c.opts.JWT == _EMPTY_ {
```

Comment:
I am not sure if my earlier comment was properly posted. Trying again:

Why don't we consolidate everything under that first part of the if statement:
```
if ws := c.ws; ws != nil {
    if c.opts.JWT == _EMPTY_ {
        c.opts.JWT = ws.cookieJwt
    }
    if c.opts.Username == _EMPTY_ {
        c.opts.Username = ws.cookieUsername
    }
    ...
}
```
Basically, no need to check for `ws := c.ws; ws != nil` every time.

Reply:
didn't complete it yet.. Now it is in